### PR TITLE
[PM-5281] [PM-5282] Disable User Interaction Post 'Always for this Site' Selection and Preserve Prior Exclusions

### PR DESCRIPTION
**Pull request by gbubemismith**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
- When the window delays closing for 2 seconds after selecting “Always for this site” you can still interact with the window. It’s possible if you’re fast enough to try and save a passkey to an item, which results in an error. If possible, we should try to disable the button while it’s in that 2-second state to show you the toast.

- When you select “Always for this site” instead of adding the domain to your existing excluded domains, it deletes all your existing excluded domains and replaces it with just that one.

## Code changes

- **apps/browser/src/vault/popup/components/fido2/fido2-use-browser-link.component.ts:** Retrieved existing excluded domains and appended the new excluded domain to the list. I added the `showOverlay` output property, which is used to show an overlay to prevent further interaction by the user when `Always for this site` is selected.
- **apps/browser/src/vault/popup/components/fido2/fido2.component.html:** Added an overlay div, when triggered, covers the fido2 component page 

## Screenshots

https://github.com/bitwarden/clients/assets/13024008/85080c2c-753b-4cc8-8dce-bd4ac817cb26

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by shane-melton — COMMENTED** [review]
This looks good! One small question -- Do you foresee any other flows that will need to disable user input in the Fido component in the future? 

If not, can we just move the overlay into the `fido2-user-browser-link` component? This would avoid the need to message passing between the parent `fido2` component via the event emitter and would be a bit more self contained I think.

**Comment by gbubemismith** [comment]
> This looks good! One small question -- Do you foresee any other flows that will need to disable user input in the Fido component in the future?
> 
> If not, can we just move the overlay into the `fido2-user-browser-link` component? This would avoid the need to message pass between the parent `fido2` component via the event emitter and would be a bit more self contained I think.

That flow wouldn't be needed in future. I would move the logic to the `fido2-user-browser-link` component. Thanks

File: apps/browser/src/vault/popup/components/fido2/fido2-use-browser-link.component.ts [hunk 1]
```diff
@@ -79,7 +79,7 @@ export class Fido2UseBrowserLinkComponent {
       return;
     }
     // Show overlay to prevent the user from interacting with the page.
-    this.showOverlay.emit();
+    this.setShowOverlay();
```
Comment (shane-melton) [inline apps/browser/src/vault/popup/components/fido2/fido2-use-browser-link.component.ts:82]: suggestion: I don't think we need the private method here, and can just set `this.isOverlay = true` directly.

**Review by shane-melton — APPROVED** [review]
Looks good to go! Thanks for making those changes!

**Pull request merged**